public/app/features/alerting/state/actions.ts

```diff
@@ -248,32 +223,41 @@ function handleJSONResponse(frames: DataFrameJSON[]) {
 }
 
 function buildDataQueryModel(queryOptions: QueryGroupOptions, defaultDataSource: DataSourceApi) {
-  return queryOptions.queries.map((query) => {
-    let dataSource: QueryGroupDataSource;
-    const isExpression = query.datasource === ExpressionDatasourceID;
-
-    if (isExpression) {
-      dataSource = { name: ExpressionDatasourceID, uid: ExpressionDatasourceID };
-    } else {
-      const dataSourceSetting = getDataSourceSrv().getInstanceSettings(query.datasource);
-
-      dataSource = {
-        name: dataSourceSetting?.name ?? defaultDataSource.name,
-        uid: dataSourceSetting?.uid ?? defaultDataSource.uid,
+  return queryOptions.queries.map((query: DataQuery) => {
```

Comment:
hmmm, I see in some functions above `buildDataQueryModel` is called with empty object `{}` cast to `QueryGroupOptions`. doesn't it then break here with `undefined does not have property map` or something similar here?

Reply:
Will remove this in a follow up PR